src/main/java/org/proshin/finapi/primitives/optional/OptionalBigDecimalOf.java

```diff
@@ -25,7 +25,7 @@
     private final Supplier<Optional<BigDecimal>> origin;
 
     public OptionalBigDecimalOf(final JSONObject origin, final String name) {
-        this(new OptionalOf<>(origin, name, JSONObject::getBigDecimal));
+        this(new OptionalOf<>(origin, name, (json, key) -> new BigDecimal(json.optString(key))));
```

Comment:
I think it's easier if we just read the value as a string and then convert it to BigDecimal as it is, since the precision is already ensured by finAPI.

Reply:
yeah, ok, only one thing: not `optString`, but `getString` as the third parameter can only be triggered if the field is actually presented in the JSON